File: bike-shop/bike-shop/backend/src/main/java/com/example/bike_shop/service/CategoryService.java
package com.example.bike_shop.service;

import com.example.bike_shop.dto.CategoryDTO;

import java.util.List;

public interface CategoryService {
    List<CategoryDTO> getAll();
    CategoryDTO getById(Long id);
    CategoryDTO create(CategoryDTO dto);
    CategoryDTO update(Long id, CategoryDTO dto);
    void delete(Long id);
}

File: bike-shop/bike-shop/frontend/src/services/categoryService.js
import axiosClient from "./axiosConfig";

const categoryService = {
  getAll: () => axiosClient.get("/categories"),
  getById: (id) => axiosClient.get(`/categories/${id}`),
  create: (data) => axiosClient.post("/categories", data),
  update: (id, data) => axiosClient.put(`/categories/${id}`, data),
  delete: (id) => axiosClient.delete(`/categories/${id}`),
};

export default categoryService;
